parsing/location.ml

```diff
@@ -225,6 +225,89 @@ let print_locs ppf locs =
   Format.pp_print_list ~pp_sep:(fun ppf () -> Format.fprintf ppf ",@ ")
     print_loc ppf locs
 
+(******************************************************************************)
+(* An interval set structure; additionally, it stores user-provided information
+   at interval boundaries.
+
+   The implementation provided here is naive and assumes the number of intervals
+   to be small, but the interface would allow for a more efficient
+   implementation if needed.
+
+   Note: the structure only stores maximal intervals (that therefore do not
+   overlap).
```

Comment:
Out of curiosity, does the "maximal intervals only" aspect matter much? (It adds some complexity in `of_intervals`, and if I understand correctly it means that some of the user-provided bound annotations are lost.)

Reply:
Right. I think it simplifies the logic for inserting `@{<..>` and `@}` later, which will get trickier if it has to handled possibly overlapping intervals (and I think the previous implementation which tried to do it was incorrect).

So I think it's better to merge intervals directly in the interval set structure, additionally that's what an actual interval set implementation would do I believe.